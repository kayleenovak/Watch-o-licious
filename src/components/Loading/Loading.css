.loading-page {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.loading-cupcake {
  margin-top: 50px;
  border-radius: 50%;
  width: 100px;
  height: 100px;
  background-color: #aea7b9;
  border: 2px solid #999;
  animation: bounce 1s infinite alternate;
  -webkit-animation: bounce 1s infinite alternate;
}

.loading-message {
  font-size: 35px;
  color: white;
}

@keyframes bounce {
  from {
    transform: translateY(0px);
  }
  to {
    transform: translateY(-15px);
  }
}
@-webkit-keyframes bounce {
  from {
    transform: translateY(0px);
  }
  to {
    transform: translateY(-15px);
  }
}