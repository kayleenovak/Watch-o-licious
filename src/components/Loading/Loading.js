import React from 'react'
import './Loading.css'

export const Loading = () => {
  return (
    <section className='loading-page'>
      <img src='/cupcake.svg' className='loading-cupcake' />
      <h2 className='loading-message'>Loading, please wait...</h2>
    </section>
  )
}