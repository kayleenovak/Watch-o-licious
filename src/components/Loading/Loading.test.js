import React from 'react'
import { Loading } from './Loading.js'
import { shallow } from 'enzyme'

describe('Loading', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Loading />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should render 4 NavLinks', () => {
    expect(wrapper.find('section').length).toEqual(1)
    expect(wrapper.find('img').length).toEqual(1)
    expect(wrapper.find('h2').length).toEqual(1)
  })
})