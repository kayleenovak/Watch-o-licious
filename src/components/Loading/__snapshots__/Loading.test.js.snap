// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="cupcake"
          className="loading-cupcake"
          src="/cupcake.svg"
        />,
        <h2
          className="loading-message"
        >
          Loading, please wait...
        </h2>,
      ],
      "className": "loading-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "cupcake",
          "className": "loading-cupcake",
          "src": "/cupcake.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading, please wait...",
          "className": "loading-message",
        },
        "ref": null,
        "rendered": "Loading, please wait...",
        "type": "h2",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="cupcake"
            className="loading-cupcake"
            src="/cupcake.svg"
          />,
          <h2
            className="loading-message"
          >
            Loading, please wait...
          </h2>,
        ],
        "className": "loading-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "cupcake",
            "className": "loading-cupcake",
            "src": "/cupcake.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading, please wait...",
            "className": "loading-message",
          },
          "ref": null,
          "rendered": "Loading, please wait...",
          "type": "h2",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
